arangod/IResearch/IResearchView.cpp

```diff
@@ -568,15 +568,13 @@ arangodb::Result IResearchView::dropImpl() {
 
   if (!stale.empty()) {
     // check link auth as per https://github.com/arangodb/backlog/issues/459
-    if (!arangodb::ExecContext::current().isSuperuser()) {
```

Comment:
If the user is a superuser then the user should have access to the collection. Therefore removing this short-cut as Superuser is not well-defined.